app/controllers/api/policies_controller.rb

```diff
@@ -3,5 +3,65 @@ class PoliciesController < BaseController
     include Subcollections::Conditions
     include Subcollections::Events
     include Subcollections::PolicyActions
+    REQUIRED_FIELDS = %w(name description towhat conditions_ids action_id event_id).freeze
+
+    def create_resource(type, _id, data = {})
+      assert_id_not_specified(data, type)
+      assert_all_required_fields_exists(data, type, REQUIRED_FIELDS)
+      create_policy(data)
+    end
+
+    def edit_resource(type, id = nil, data = {})
+      raise BadRequestError, "Must specify an id for editing a #{type} resource" unless id
+      policy = resource_search(id, type, collection_class(:policies))
+      add_associations(data, policy)
+      policy.update_attributes(data)
+      policy
```

Comment:
let's add a begin/rescue block for add_associations up to policy and raise the BadRequestError, "Could not update the policy - #{err}"